src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java

```diff
@@ -0,0 +1,154 @@
+package it.pagopa.ecommerce.commons.client;
+
+import it.pagopa.ecommerce.commons.exceptions.NpgResponseException;
+import it.pagopa.ecommerce.commons.generated.npg.v1.ApiClient;
+import it.pagopa.ecommerce.commons.generated.npg.v1.api.PaymentServicesApi;
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+
+@ExtendWith(MockitoExtension.class)
+class NpgClientTests {
+    @Mock
+    private ApiClient apiClient;
+    @Mock
+    @Qualifier("npgWebClient")
+    private PaymentServicesApi paymentServicesApi;
+
+    private NpgClient npgClient;
+
+    @BeforeEach
+    public void init() {
+        Mockito.when(paymentServicesApi.getApiClient()).thenReturn(apiClient);
+        npgClient = new NpgClient(paymentServicesApi, "xxx");
```

Comment:
You can make this a constant to make tests more readable

```suggestion
        npgClient = new NpgClient(paymentServicesApi, MOCKED_API_KEY);
```